sdk/cognitiveservices/Vision.CustomVision.Prediction/src/Customizations/ApiKeyServiceClientCredentials.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Microsoft.Azure.CognitiveServices.Vision.CustomVision.Prediction
+{
+    using System;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest;
+
+    /// <summary>
+    /// Allows authentication to the API using a basic apiKey mechanism
+    /// </summary>
+    public class ApiKeyServiceClientCredentials : ServiceClientCredentials
+    {
+        private readonly string predictionKey;
+
+        /// <summary>
+        /// Creates a new instance of the ApiKeyServiceClientCredentails class
+        /// </summary>
+        /// <param name="predictionKey">The prediction key to authenticate and authorize as</param>
+        public ApiKeyServiceClientCredentials(string predictionKey)
+        {
+            this.predictionKey = predictionKey;
+        }
+
+        /// <summary>
+        /// Add the Basic Authentication Header to each outgoing request
+        /// </summary>
+        /// <param name="request">The outgoing request</param>
+        /// <param name="cancellationToken">A token to cancel the operation</param>
+        public override Task ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+                throw new ArgumentNullException("request");
+
+            request.Headers.Add("Prediction-Key", this.predictionKey);
+
+            return Task.FromResult<object>(null);
```

Comment:
Does this mean that this will just give back a null object back?

Reply:
Correct, I'm following the pattern the other ApiKeyServiceClientCredentials are using such as: https://github.com/Azure/azure-sdk-for-net/blob/4ac320b7bb143145d67d396cd33cf9cc1dc21a2f/sdk/cognitiveservices/Vision.ComputerVision/src/Customizations/ApiKeyServiceClientCredentials.cs